### ISSUE TITLE
v7

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#399]
- [ ] Add Azure to recommened preset if it is not false positive https://github.com/secretlint/secretlint/pull/396

**Comment by azu** [comment]
## Changes

- Convert to [Pure ESM package](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c)
  - If your scripts are CommonJS, you need to dynamic import `import("@secretlint/core")`.
  - For command line/Docker user, No change anything!
- Convert `@secretlint/types` to dual package
  - [Dual CommonJS/ES module packages](https://nodejs.org/api/packages.html#dual-commonjses-module-packages)
  - You can import `@secretlint/types` from both CJS and ESM.
- Add new "json" formatter for secretlint
  - output format is changed from old "json" formatter
- @secretlint/tester is migrated to ESM, But we can load it using dynamic import from CJS
  - Use `node:test` instead of Mocha
  - Support URL object in `snapshotDirectory` option.

```diff
- import { snapshot } from "@secretlint/tester");
import path from "path";
import { creator as rule } from "../src/index";
+ import test from "node:test";
+ (async function () {
+   const snapshot = (await import("@secretlint/tester")).snapshot;
-    describe("@secretlint/secretlint-rule-preset-canary", () => {
+    test("@secretlint/secretlint-rule-preset-canary", async (t) => {
-        snapshot({
+        return snapshot({
            defaultConfig: {
                rules: [
                    {
                        id: "@secretlint/secretlint-rule-preset-canary",
                        rule,
                        rules: [],
                        options: {},
                    },
                ],
            },
            updateSnapshot: !!process.env.UPDATE_SNAPSHOT,
            snapshotDirectory: path.join(__dirname, "snapshots"),
        }).forEach((name, test) => {
-            it(name, async function () {
+            return it(name, async (context) => {
                const status = await test();
                if (status === "skip") {
-                    this.skip();
+                    context.skip();
                }
            });
        });
    });
+});
```

**Comment by azu** [comment]
- [Release v7.0.0 · secretlint/secretlint](https://github.com/secretlint/secretlint/releases/tag/v7.0.0)
- https://efcl.info/2023/07/05/secretlint-v7/
